packages/jest-cli/src/search_source.js

```diff
@@ -209,7 +209,9 @@ export default class SearchSource {
     } else if (globalConfig.findRelatedTests && paths && paths.length) {
       return Promise.resolve(this.findRelatedTestsFromPattern(paths));
     } else {
-      const validTestPaths = paths && paths.filter(fs.existsSync);
+      const validTestPaths =
+        paths &&
+        paths.filter(fs.existsSync).filter(name => fs.lstatSync(name).isFile());
```

Comment:
What about
```js
    const validTestPaths =
        paths &&
        paths.filter(name => {
            try {
                return fs.lstatSync(name).isFile();
            } catch (e) {
                return false;
            }
        });
```

to at least save a single FS operation?

Reply:
Good call; fixed